src/nisarqa/utils/lonlat.py

```diff
@@ -2,52 +2,93 @@
 
 import os
 import textwrap
-from collections.abc import Sequence
+from collections.abc import Generator, Iterable, Sequence
 from dataclasses import dataclass
 from pathlib import Path
+from typing import Any
+
+import numpy as np
 
 import nisarqa
 
 objects_to_skip = nisarqa.get_all(__name__)
 
 
-def unwrap_longitudes(lon_lat_points: Sequence[LonLat]) -> list[LonLat]:
+def unwrap_longitudes(lons: Iterable[float]) -> list[float]:
     """
-    Normalize longitudes so that they form a continuous quadrilateral
-    across the antimeridian.
+    Unwrap longitudes if the values cross the antimeridian.
+
+    Specifically, longitudes are normalized so that the absolute difference
+    between any adjacent pair of longitude values is <= 180 degrees.
+
+    Of note, in the case of the an antimeridian crossing, longitude values
+    are "unwrapped" to extend beyond the interval of +/-180 degrees
+    for the crossing.
 
     Arguments
     ---------
-    lon_lat_points : Sequence of nisarqa.LonLat
-        List of nisarqa.LonLat (in degrees)
+    lons : Sequence of float
+        Sequence of longitude values (in degrees).
 
     Returns
     -------
-    unwrapped_lon_lat : list of nisarqa.LonLat
-        Copy of `lon_lat`, but in the case of an antimeridian crossing
-        the longitude values are "unwrapped" to extend beyond the
-        interval of +/-180 degrees. The ordering of the points is preserved.
+    unwrapped : list of nisarqa.LonLat
+        Copy of `lons`, but unwrapped so that the absolute difference
+        between any adjacent pair of longitude values is <= 180 degrees.
+        The ordering of the points is preserved.
     """
-    unwrapped = [lon_lat_points[0]]
 
-    for i in range(1, len(lon_lat_points)):
-        prev_lon = lon_lat_points[i - 1].lon
-        current_lon = lon_lat_points[i].lon
+    # since `lons` is an Iterable, `wrap_to_interval` returns a list
+    lons_360 = nisarqa.wrap_to_interval(val=lons, start=-360, stop=360)
```

Comment:
Let's initially wrap the longitudes to [-180, 180) instead of [-360, 360).

The code will still work as long as the longitude values are within [-360, 360), but the common convention is that longitude values should be within [-180, 180] (except in the case of an antimeridian crossing).

```suggestion
    lons_180 = nisarqa.wrap_to_interval(val=lons, start=-180, stop=180)
```

Reply:
Note this might require some updates to the docstrings as well.